lib/tasks/batch_import.rb

```diff
@@ -0,0 +1,117 @@
+#
+# Ruby process for batch ingest, called by import_batches rake task
+#
+
+def import_batch
+  manifest_filename = "manifest.yml"
+  ingest_dir = "spec/fixtures/ingest/pmp"
+  batch_folders = Dir.glob(ingest_dir + "/*").select { |f| File.directory? f }
+  batch_folders.each_with_index do |subdir, index|
+    puts "Processing batch directory #{index + 1} of #{batch_folders.size}: #{subdir}"
+    begin
+      manifest = YAML.load_file(subdir + "/" + manifest_filename)
+    rescue
+      puts "ABORTING: No manifest file found."
+      manifest = nil
+    end
+    import_manifest(subdir, manifest) unless manifest.nil?
+  end
+end
+
+def import_manifest(subdir, manifest)
+  puts "Manifest file found."
+  if manifest["pageds"].nil? or manifest["pageds"].empty?
+    puts "ABORTING: No paged documents listed in manifest."
+  else
+    manifest["pageds"].each do |paged|
+      import_paged(subdir, paged)
+    end
+  end
+end
+
+def import_paged(subdir, paged_yaml)
+  paged_attributes = {}
+  begin
+    paged_yaml["descMetadata"].each_pair do |key, value|
+      paged_attributes[key.to_sym] = value
+    end
+  rescue
+    puts "ABORTING PAGED CREATION: invalid structure in descMetadata"
+  end
+  if paged_attributes.any?
+    begin
+      paged = Paged.new(**paged_attributes)
```

Comment:
Don't really need the double-splat here:  there's nothing for paged_attributes to be merged with.